src/internal/utils/sort.ts

```diff
@@ -108,10 +109,26 @@ function getDefaultBarChartSort(
     resultSpec: AFM.IResultSpec,
     canSortStackTotalValue: boolean = false,
 ): AFM.SortItem[] {
+    const measureItemsCount: number = get(afm, "measures", []).length;
+    const viewByDimensionItems = getDimensionItems(resultSpec, VIEW_BY_DIMENSION);
     const viewByAttributeIdentifier = getFirstAttributeIdentifier(resultSpec, VIEW_BY_DIMENSION);
     const stackByAttributeIdentifier = getFirstAttributeIdentifier(resultSpec, STACK_BY_DIMENSION);
 
-    if ((viewByAttributeIdentifier && stackByAttributeIdentifier) || canSortStackTotalValue) {
+    if (viewByDimensionItems.length >= 2) {
```

Comment:
I think it could be greater than 2, right? just 2 or <2
so how about? `if (viewByDimensionItems.length === 2) {...}`